client/llb/source.go

```diff
@@ -201,7 +201,7 @@ func Git(remote, ref string, opts ...GitOption) State {
 	url := ""
 
 	for _, prefix := range []string{
-		"http://", "https://", "git://", "git@",
```

Comment:
Why?

Reply:
Without this change, when I do
```
llb.Git("git@github.com:earthly/earthly.git", "main")
```

it causes the string to get trimmed into `github.com:earthly/earthly.git`, which then causes buildkit to attempt to parse the string as if it were a URL (in https://github.com/moby/buildkit/pull/1782/files#diff-dc3734c7af40d455fbd22d64b372bf572aa7314209bafc3d5f1f809203186d6eL27 )

which ultimately gives me the error: `failed to load cache key: parse https://github.com:earthly/earthly.git: invalid port ":earthly" after host`.

I wonder if there's a way we could reproduce this error via an integration test to show that this fixes the issue? I don't know enough about buildkit's integrating testing to add one (but would be keen on learning with some help).

Reply:
This is the reason why `AttrFullRemoteURL` was added I believe that should be used instead to get this info.

Reply:
I'll back out this change, and try using `AttrFullRemoteURL` in a follow up PR.